apps/test/util/testUtils.js

```diff
@@ -462,38 +447,46 @@ export function clearTimeoutsBetweenTests() {
  *   suite.  This should normally be 'false' to avoid unneeded impact on the
  *   runtime of the suite, but setting it 'true' is very useful for isolating
  *   the test that's causing related failures.
+ * @param {function} inScope callback function containing the tests to run
+ *   with the body cleanup check in place.
  */
-export function enforceDocumentBodyCleanup({checkEveryTest = false}) {
+export function enforceDocumentBodyCleanup({checkEveryTest = false}, inScope) {
   let initialInnerHTML;
   const beforeFn = checkEveryTest ? beforeEach : before;
   const afterFn = checkEveryTest ? afterEach : after;
 
-  beforeFn(() => {
-    if (!initialInnerHTML) {
-      initialInnerHTML = document.body.innerHTML;
-    }
-    sinon.spy(document.body, 'addEventListener');
-    sinon.spy(document.body, 'removeEventListener');
-  });
+  // The additional describe calls ensure correct ordering of the
+  // before/after steps relative to other test steps.
+  describe('', () => {
+    beforeFn(() => {
+      if (!initialInnerHTML) {
+        initialInnerHTML = document.body.innerHTML;
+      }
+      sinon.spy(document.body, 'addEventListener');
+      sinon.spy(document.body, 'removeEventListener');
+    });
 
-  afterFn(() => {
-    if (initialInnerHTML !== document.body.innerHTML) {
-      throw new Error(
-        'Test modified document.body.innerHTML:' +
-        '\n\nInitial:\n' +
-        initialInnerHTML +
-        '\n\nAfter:\n' +
-        document.body.innerHTML
-      );
-    }
+    afterFn(() => {
+      if (initialInnerHTML !== document.body.innerHTML) {
+        throw new Error(
+          'Test modified document.body.innerHTML:' +
+          '\n\nInitial:\n' +
+          initialInnerHTML +
+          '\n\nAfter:\n' +
+          document.body.innerHTML
+        );
+      }
 
-    if (document.body.addEventListener.callCount !== document.body.removeEventListener.callCount) {
-      throw new Error(
-        'Added ' + document.body.addEventListener.callCount + ' event listener(s)' +
-        ' to document.body, but only removed ' + document.body.removeEventListener.callCount + ' listeners'
-      );
-    }
-    document.body.addEventListener.restore();
-    document.body.removeEventListener.restore();
+      if (document.body.addEventListener.callCount !== document.body.removeEventListener.callCount) {
+        throw new Error(
+          'Added ' + document.body.addEventListener.callCount + ' event listener(s)' +
+          ' to document.body, but only removed ' + document.body.removeEventListener.callCount + ' listeners'
+        );
+      }
+      document.body.addEventListener.restore();
+      document.body.removeEventListener.restore();
```

Comment:
are we at risk from double-restoring these if enforceDocumentBodyCleanup is called twice around the same test case? if we are safe from this, could be worth documenting why/how

Reply:
At least in this case, the `spy` calls in the `beforeFn` would fail because re-wrapping a spy or stub is not allowed by sinon.